src/llmcompressor/modifiers/quantization/gptq/base.py

```diff
@@ -174,14 +192,17 @@ def on_start(self, state: State, event: Event, **kwargs):
 
         # register gptq hooks
         added_hook = False
-        for module in state.model.modules():
-            if getattr_chain(module, "quantization_scheme.weights", None) is not None:
-                # HACK: previously, embeddings were not quantized because they were not
-                # accessible by the layer compressor. For now, we manually ignore it,
-                # but in the FUTURE this should be ignored by the user
-                if not isinstance(module, torch.nn.Embedding):
-                    self.register_hook(module, self.calibrate_module, "forward")
-                    added_hook = True
+        for name, module in match_named_modules(state.model, self.targets, self.ignore):
+            assert (
+                getattr_chain(module, "quantization_scheme.weights", None) is not None
+            ), f"Matched target {name} does not have associated weight quantization"
```

Comment:
Can we use if/raise instead of assert?

Generally I think that's preferred for production code because asserts can be removed / ignored by the end user.

Reply:
i ended up removing completely. `match_named_modules` is returning non-quantizable layers like `self_attn` (in addition to `self_attn.q_proj) and we need to skip them here